### PR TITLE
[FIX] account: Reverse tags sign when using reconciliation model 

### PR DESCRIPTION
**Pull request by abz-odoo**
When reconciling bank statement using manual operation (write-off), the tags generated on the journal entry have their sign reversed between manual creation (using the account and tax field in the reconciliation widget) and automated creation using a button (created from reconciliation model).

This pr adds the inversion of the tags in the case of the use of a reconciliation model

opw-2540090

### CODE REVIEW

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 1]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+            'refund_repartition_line_ids': [
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'base',
+                    'tag_ids': [(6, 0, tax_tag_neg.ids)],#-G1
+                }),
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                    'tag_ids': [(6, 0, tax_tag_pos.ids)],#-G1,+GST
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1031]: remove comments

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 2]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+        country = self.env.ref('base.us')
+        tax_report = self.env['account.tax.report'].create({
+            'name': "Tax report",
+            'country_id': country.id,
+        })
+        tax_report_line = self.env['account.tax.report.line'].create({
+            'name': 'test_tax_report_line',
+            'tag_name': 'test_tax_report_line',
+            'report_id': tax_report.id,
+            'sequence': 10,
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1000]: sequence is useless

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 3]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+        })
+        tax_tag_pos = tax_report_line.tag_ids.filtered(lambda x: not x.tax_negate)
+        tax_tag_neg = tax_report_line.tag_ids.filtered(lambda x: x.tax_negate)
+        tax = self.env['account.tax'].create({
+            'name': 'GST Inc Sale (10%)',
+            'type_tax_use': 'sale',
+            'amount': 10,
+            'amount_type': 'percent',
+            'price_include': True,
+            'invoice_repartition_line_ids': [
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1010]: remove `price_include`, `amount_type`, `type_tax_use` and rename the tax to something like `test tax`

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 4]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+        })
+        reconciliation_model = self.env['account.reconcile.model'].create({
+            'name': 'Charge with Tax',
+            'company_id': self.company.id,
+            'line_ids': [(0, 0, {
+                'company_id': self.company.id,
+                'account_id': self.company_data['default_account_expense'].id,
+                'amount_type': 'percentage',
+                'amount_string': '100',
+                'tax_ids': [(6, 0, [tax.id])]
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1043]: `tax.ids`

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 5]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+        })
+        bank_stmt = self.env['account.bank.statement'].create({
+            'company_id': self.company.id,
+            'journal_id': self.bank_journal.id,
+            'date': time.strftime('%Y-07-15'),
+            'name': 'test',
+            'line_ids': [(0, 0, {
+                'payment_ref': 'testLine',
+                'amount': 5,
+                'date': time.strftime('%Y-07-15'),
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1054]: put `2020-07-15` directly

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 6]
```diff
@@ -1,5 +1,6 @@
 # -*- coding: utf-8 -*-
 from freezegun import freeze_time
+import time
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:3]: remove this import

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 7]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+                    'repartition_type': 'tax',
+                    'tag_ids': [(6, 0, tax_tag_pos.ids)],#-G1,+GST
+                }),
+            ],
+        })
+        reconciliation_model = self.env['account.reconcile.model'].create({
+            'name': 'Charge with Tax',
+            'company_id': self.company.id,
+            'line_ids': [(0, 0, {
+                'company_id': self.company.id,
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1039]: default company is the current one so you can remove `company_id` here and above

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 8]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+        bank_stmt = self.env['account.bank.statement'].create({
+            'company_id': self.company.id,
+            'journal_id': self.bank_journal.id,
+            'date': time.strftime('%Y-07-15'),
+            'name': 'test',
+            'line_ids': [(0, 0, {
+                'payment_ref': 'testLine',
+                'amount': 5,
+                'date': time.strftime('%Y-07-15'),
+                'partner_name': 'test',
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1055]: `partner_name`, `company_id` are useless

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 9]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+            })]
+        })
+        res = reconciliation_model._get_write_off_move_lines_dict(bank_stmt.line_ids[0], 5)
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1058]: you only have one st_line so `bank_stmt.line_ids` is enough

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 10]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+                'debit': 0.45,
+                'credit': 0,
+                'analytic_account_id': False,
+                'analytic_tag_ids': False,
+                'tax_exigible': 'on_invoice',
+                'tax_repartition_line_id': tax.invoice_repartition_line_ids[1].id,
+                'tax_ids': [(6, 0, [])],
+                'tax_tag_ids': [(6, 0, tax.refund_repartition_line_ids[1].tag_ids.ids)],
+                'currency_id': False,
+                'reconcile_model_id': reconciliation_model.id,
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1087]: you can remove `currency_id`, `debit`, `credit`, `analytic_account_id`, `analytic_tag_ids`, `tax_exigible`
Also, plz use the same order in both dictionaries ><

File: addons/account/tests/test_reconciliation_matching_rules.py [hunk 11]
```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
...
+            }
+        ]
+        self.assertEqual(res, model, 'Tags are not reverted')
```
Comment (smetl) [inline addons/account/tests/test_reconciliation_matching_rules.py:1090]: use `assertRecordValues` plz

**Review by smetl — CHANGES_REQUESTED** [review]

**Review by smetl — APPROVED** [review]
@robodoo r+

**Comment by abz-odoo** [comment]
@robodoo r+

**Comment by abz-odoo** [comment]
@robodoo retry

**Comment by abz-odoo** [comment]
linked pr: https://github.com/odoo/enterprise/pull/19979

**Pull request closed without merging**